torch_xla/core/xla_model.py

```diff
@@ -691,17 +692,34 @@ def add_step_closure(closure, args=()):
   Args:
     closure (callable): The function to be called.
     args (tuple): The arguments to be passed to the closure.
+    run_async: If True, run the closure asynchronously
   """
   devctx = _get_device_context()
-  step_closures = getattr(devctx, 'step_closures', None)
-  if step_closures is None:
-    step_closures = []
-    devctx.step_closures = step_closures
-  step_closures.append(lambda a=args: closure(*a))
+  if not run_async:
+    step_closures = getattr(devctx, 'step_closures', None)
+    if step_closures is None:
+      step_closures = []
+      devctx.step_closures = step_closures
+    step_closures.append(lambda a=args: closure(*a))
+  else:
+    async_step_closures = getattr(devctx, 'async_step_closures', None)
+    if async_step_closures is None:
+      async_step_closures = []
+      devctx.async_step_closures = async_step_closures
+    async_step_closures.append(lambda a=args: closure(*a))
```

Comment:
Can we dedup this part with something like 
```
    closures_type = 'async_step_closures' if run_async else 'step_closures'
    step_closures = getattr(devctx, closures_type, None)
    if step_closures is None:
      step_closures = []
      setattr(devctx, closures_type, step_closures)
    step_closures.append(lambda a=args: closure(*a))
```

Reply:
done